jstests/aggregation/bugs/server7695_isodates.js

```diff
@@ -0,0 +1,269 @@
+// SERVER-7695: Add $isoWeek, $isoWeekYear, and $isoDayOfWeek aggregation expressions.
+
+(function() {
+    "use strict";
+    var coll = db.server7695;
+    var testOpCount = 0;
+
+    coll.drop();
+
+    // Seed collection so that the pipeline will execute.
+    assert.writeOK(coll.insert({}));
+
+    // Helper for testing that op returns expResult.
+    function testOp(op, value, expResult) {
+        testOpCount++;
+        var pipeline = [{$project: {_id: 0, result: {}}}];
+        pipeline[0].$project.result[op] = value;
+        var msg = "Exptected {"+op+": "+value+"} to equal: "+expResult;
+        assert.eq(
+            coll.aggregate(pipeline).toArray()[0].result,
+            [{result: expResult}][0].result,
+            msg);
+    }
+
+    // while development there was a bug which caused an error with $dateToString if the order of
+    // %V and %G changed, so I added this test to prevent regression
+    testOp('$dateToString', { date: new Date("1900-12-31T23:59:59Z"), format: "%V-%G"}, "01-1901")
+    // this is failing, but it shouldn't as it is the same as above, only rotated
```

Comment:
Edit: I removed the log line, and it started failing for me again. I think I may have found the source, but I'm still unsure why it's happening:
```
Process 15116 stopped
* thread #2: tid = 0x4c5827, 0x000000010097f0be mongod`mongo::ExpressionIsoWeek::lastWeek(year=1900) + 62 at expression.cpp:3181, stop reason = step over
    frame #0: 0x000000010097f0be mongod`mongo::ExpressionIsoWeek::lastWeek(year=1900) + 62 at expression.cpp:3181
   3178	    tm.tm_hour = 23;
   3179	    tm.tm_min = 59;
   3180	    tm.tm_sec = 59;
-> 3181	    mktime(&tm);
   3182	
   3183	    // From: https://en.wikipedia.org/wiki/ISO_week_date#Last_week :
   3184	    // If 31 December is on a Monday, Tuesday or Wednesday, it is in week 01 of
(lldb) n
2016-03-14T11:13:55.799-0400 I STORAGE  [DataFileSync] flushing mmaps took 178685ms  for 5 files
2016-03-14T11:13:55.799-0400 I NETWORK  [initandlisten] connection accepted from 127.0.0.1:52490 #2 (2 connections now open)
Process 15116 stopped
* thread #2: tid = 0x4c5827, 0x000000010097f0c6 mongod`mongo::ExpressionIsoWeek::lastWeek(year=1900) + 70 at expression.cpp:3191, stop reason = step over
    frame #0: 0x000000010097f0c6 mongod`mongo::ExpressionIsoWeek::lastWeek(year=1900) + 70 at expression.cpp:3191
   3188	    // 52 (or 53 if the year just ending is a leap year); if on a Saturday or
   3189	    // Sunday, it is in week
   3190	    // 52 of the year just ending.
-> 3191	    if (tm.tm_wday > 0 && tm.tm_wday < 4) {  // Mon(1), Tue(2), and Wed(3)
   3192	        return 1;
   3193	    } else if (tm.tm_wday == 4) {  // Thu (4)
   3194	        return 53;
(lldb) p tm.tm_wday
(int) $17 = 6938768
(lldb) p tm
(tm) $18 = {
  tm_sec = 59
  tm_min = 59
  tm_hour = 23
  tm_mday = 31
  tm_mon = 11
  tm_year = 0
  tm_wday = 6938768
  tm_yday = 28672
  tm_isdst = 1
  tm_gmtoff = 25769803782
  tm_zone = 0x0000000700000005 ""
}
```

I'll continue to investigate.